### DISCUSSION TITLE
Question: Do Rbatis support embedded struct?

### DISCUSSION THREAD

**Discussion by j4w0** [rbatis/rbatis#282] [General]
Recently I am using rbatis in my experimental project. But I've found that the doc tells little about the model definition.
So, Any support about embedded struct? It will be more convenient if it does.

**Comment by zhuxiujia** [comment]
> Recently I am using rbatis in my experimental project. But I've found that the doc tells little about the model definition.
> So, Any support about embedded struct? It will be more convenient if it does.

you can see our doc website https://rbatis.github.io/rbatis.io/#/en/

**Comment by j4w0** [comment]
Well I have read the doc but I am still confuse in embedding a struct into another struct.
Like This:
```rust
struct EmbeddedStruct {
  pub random_id: uint64
}
struct MainStruct {
  //...
  pub embedded_struct: EmbeddedStruct
}
```
Is it possible?

**Comment by zhuxiujia** [comment]
> Well I have read the doc but I am still confuse in embedding a struct into another struct.
> Like This:
> 
> ```rust
> struct EmbeddedStruct {
>   pub random_id: uint64
> }
> struct MainStruct {
>   //...
>   pub embedded_struct: EmbeddedStruct
> }
> ```
> 
> Is it possible?

RBatis uses JSON to encode and decode data and currently does not support EmbeddedStruct

**Comment by j4w0** [comment]
Well thanks a lot!
It will be a great feature if rbatis support it as it can make a model reusable. Please consider to add this in roadmap!
